docs/content/1.documentation/5.advanced/3.strict-csp.md

```diff
@@ -0,0 +1,657 @@
+
+::alert{type="success"}
+In summary, Nuxt applications are perfectly capable of running with Strict CSP under two conditions:
+  1. You use `'strict-dynamic'` on the `script-src` policy
+  2. You use `'useHead'` to insert external scripts
+::
+
+With this setup, a Level 3 configuration file is much simpler:
+```ts
+export defaultNuxtConfig({
+  security: {
+    headers: {
+      contentSecurityPolicy: {
+        "script-src": [
+          "'nonce-{{nonce}}'",
+          // The nonce allows the root script
+          "'strict-dynamic'" 
+          // All scripts inserted by the root script will also be allowed
+        ]
+      }
+    }
+  }
+})
+```
+
+## Whitelisting external resources
+
+### The `useHead` composable
+
+`useHead` allows you to insert scripts in the `<head>` tag of your HTML document.
+It is an isomorphic function, i.e. it runs transparently both on the server-side and the client-side.
+In addition, it is executed by the Nuxt root script, which is already authorized under `'strict-dynamic'`.
+
+It is therefore the best place to insert external scripts for your app.
+
+```ts
+useHead({
+  script: [
+    { src: 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js',
+      crossorigin: 'anonymous'
+    },
+    // Insert any external script with strict-dynamic
+    { textContent: "console.log('Hello World')" }
+    // Insert an inline script
+  ]
+})
+```
+
+With `'strict-dynamic'`, these scripts will be allowed as children of the Nuxt root script.
+
+Alternatively, if you cannot use `useHead`, you can also insert tags directly in the DOM (e.g. via `document.createElement()`). As long as the function is invoked within your Nuxt application, `'strict-dynamic'` will allow execution.
+
+
+### SSR mode
+
+
+In SSR mode, Nuxt Security will whitelist your scripts automatically by inserting the nonce on the server-side. 
+
+Because nonces can be used to whitelist external resources, your scripts will be allowed without any additional configuration on your part if the `nonce` option is set to `true`.
+
+::alert{type="success"}
+In SSR mode, `useHead` will always allow you to load your external scripts under Strict CSP.
+::
+
+### SSG mode
+
+In SSG mode however, there is no nonce. Because `'strict-dynamic'` cancels external whitelists, you will need to insert an `integrity` attribute to each of your external scripts in order to allow them:
+
+```ts
+useHead({
+  script: [
+    { src: 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js',
+      crossorigin: 'anonymous',
+      integrity: 'sha384-C6RzsynM9kWDrMNeT87bh95OGNyZPhcTNXj1NW7RuBCsyN/o0jlpcV8Qyq46cDfL' // Add the integrity hashes for your external scripts
+    }
+  ]
+})
+```
+Nuxt Security will be clever enough to pick up the integrity attributes and add them to the `script-src` policy if the `ssg: hashScripts` option is set to `true`.
+
```

Comment:
I am leaving out important details at this stage on Subresource Integrity, but I will include them as soon as we merge the SRI feature.
It is a critical component that needs to be included to ensure efficient whitelisting in SSG mode